File: src/server/ai/providers/groq.ts
import { AIAssistantConfig } from "../config";
import { OpenAICompatibleProvider } from "./base";

export const groqProvider = new OpenAICompatibleProvider(
  "groq",
  {
    apiKey: AIAssistantConfig.groqApiKey,
    baseUrl: "https://api.groq.com/openai/v1/chat/completions",
    model: AIAssistantConfig.groqModel,
    timeoutMs: AIAssistantConfig.requestTimeoutMs,
    maxRetries: AIAssistantConfig.maxRetries,
  }
);

File: ai-assistant/providers/groq.py
import os
from .base import OpenAICompatibleProvider

groq_provider = OpenAICompatibleProvider(
    name="groq",
    api_key=os.getenv("GROQ_API_KEY", ""),
    base_url="https://api.groq.com/openai/v1/chat/completions",
    model=os.getenv("GROQ_MODEL", "openai/gpt-oss-120b"),
)
